controllers/openstackmachine_controller.go

```diff
@@ -414,11 +458,13 @@ func (r *OpenStackMachineReconciler) getOrCreate(logger logr.Logger, cluster *cl
 		if err != nil {
 			err = errors.Errorf("machine spec is invalid: %v", err)
 			handleUpdateMachineError(logger, openStackMachine, err)
+			conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InvalidMachineSpecReason, clusterv1.ConditionSeverityError, err.Error())
 			return nil, err
 		}
 
 		instanceStatus, err = computeService.CreateInstance(openStackMachine, openStackCluster, instanceSpec, cluster.Name)
 		if err != nil {
+			conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceCreateFailedReason, clusterv1.ConditionSeverityError, err.Error())
```

Comment:
Condition severity Warning might be better here, as this is very likely to be a retryable error.